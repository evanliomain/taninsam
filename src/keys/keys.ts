/**
 * @module object=>array
 */
/**
 * This method extract own enumerable properties of an object into an array
 * See the [native Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) for more informations.
 * @return the function to apply on the object to extract its keys
 * @example
 * ```
 * keys()({ a: 1, b: 2, c: 3 }) // ['a', 'b', 'c']
 * ```
 * @example Using the chain
 * ```
 * chain({ a: 1, b: 2, c: 3 })
 *   .chain(keys())
 *   .value() // ['a', 'b', 'c']
 * ```
 */
export function keys(): (ob: Readonly<object>) => string[] {
  return (ob: Readonly<object>) => Object.keys(ob);
}
