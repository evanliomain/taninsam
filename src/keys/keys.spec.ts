import { describe, expect, test } from 'vitest';
import { keys } from './keys';

describe('keys function', () => {
  const input = { a: 1, b: 2, c: 3 };

  test('is a pure function', () => {
    keys()(input);
    expect(input).toMatchSnapshot();
  });

  test('{ a: 1, b: 2, c: 3 } |> keys === ["a", "b", "c"]', () => {
    expect(keys()(input)).toMatchSnapshot();
  });
});
