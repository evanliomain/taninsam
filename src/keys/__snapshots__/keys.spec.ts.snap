// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`keys function > { a: 1, b: 2, c: 3 } |> keys === ["a", "b", "c"] 1`] = `
[
  "a",
  "b",
  "c",
]
`;

exports[`keys function > is a pure function 1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;
